compiler/ir/ir.psi2ir/src/org/jetbrains/kotlin/psi2ir/intermediate/IrUtils.kt

```diff
@@ -46,7 +46,8 @@ fun CallReceiver.adjustForCallee(callee: CallableMemberDescriptor): CallReceiver
                         dispatchReceiverValue != null && callee.dispatchReceiverParameter == null -> dispatchReceiverValue
                         else -> extensionReceiverValue
                     }
-                withDispatchAndExtensionReceivers(newDispatchReceiverValue, newExtensionReceiverValue)
+                // TODO: Create new context receiver values
```

Comment:
This adjustment is used for synthetic properties (Java property accessor methods `getFoo` and `setFoo` used with property access syntax in Kotlin, and are treated as extension properties). 
Since Java methods can't have context receivers, you can actually put an assertion here:
```
assert(contextReceiverValues.isEmpty()) { 
    "Synthetic property accessor call adjustment with context receivers, callee: $callee"
}
```
BTW, what would happen in case of mixed Kotlin-Java hierarchy (member extension property with context receiver declared in `Base` class in Kotlin, `Child` class extends `Base` in Java)?

Reply:
Added the assertion. [See the commit.](https://github.com/JetBrains/kotlin/pull/4488/commits/ecce73b0d37dddddece99af1178fae5bace4bb34)

> What would happen in case of mixed Kotlin-Java hierarchy (member extension property with context receiver declared in Base class in Kotlin, Child class extends Base in Java)?

Let me check.

Reply:
Are you sure it is used only for synthetic properties? It looks like the function is called for regular properties too, so I need to remove the assertion.

Reply:
For example, it is called from `org.jetbrains.kotlin.psi2ir.generators.CallGenerator.generatePropertyGetterCall(CallGenerator.kt:225)`

Reply:
Yes, you're right. Ok, let's keep it as is.